docs/extend/modular-resources/create/_index.md

```diff
@@ -515,6 +515,201 @@ Expand the [Go module code](#code-a-main-entry-point-program) to view <file>main
+  </tr>
+  <tr>
+    <td><code>entrypoint</code></td>
+    <td>string</td>
+    <td><strong>Required</strong></td>
+    <td>The full filesystem path to the module file that starts your program.</td>
+  </tr>
+</table>
+
+   For example, the following represents the configuration of an example `my-module` module in the `acme` namespace:
+
+   ```json {class="line-numbers linkable-line-numbers"}
+   {
+     "name": "acme:my-module",
+     "visibility": "private",
+     "url": "https://github.com/acme-co/my-module",
+     "description": "An example custom module.",
+     "models": [
+       {
+         "api": "rdk:component:generic",
+         "model": "acme:demo:my-model"
+       }
+     ],
+     "entrypoint": "/usr/local/bin/my-module"
+   }
+   ```
+
+  {{% alert title="Important" color="note" %}}
+  If you are publishing a public module (`visibility: "public"`), the [namespace of your model](/extend/modular-resources/key-concepts/#namespace-1) must match the [namespace of your organization](/extend/modular-resources/key-concepts/#namespace). In the example above, the model namespace is set to `acme` to match the owning organization's namespace.
+  {{% /alert %}}
+
+1. Run `viam module update` to register the configuration changes you just made to `meta.json`:
+
+   - To register a *public* module, run the following command from within the same directory as your `meta.json` file:
+
+      ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+      viam module update
+      ```
+
+   - To register a *private* module, run the following command from within the same directory as your `meta.json` file, providing your organization ID:
+
+      ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+      viam module update --org-id <org-id>
+      ```
+
+   On successful update, the command will return a link to the updated module in the Viam Registry.
+
+1. Package your custom module to get it ready to upload to the Viam Registry. Currently, the Registry only supports `tar.gz` format. Run the following command to compress your custom module as a `tar.gz` archive:
+
+   ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+   tar -zcf <module-name>.tar.gz <module-name>
+   ```
+
+   Where `module-name` is the directory that contains your module.
+
+1. Run `viam module upload` to upload the updated custom module to the Viam Registry:
+
+   ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+   viam module upload --version <version> --platform <platform> <packaged-module.tar.gz>
+   ```
+
+   Where:
+
+   - `version` - provide a version for your custom module, using [semantic versioning](https://semver.org/) (example: `1.0.0`). The Viam Registry does not perform any validation on this value. You can later increment this value with subsequent `viam module upload` commands.
```

Comment:
Assuming we should apprise users that versioning in entirely on their end. Please let me know if we do in fact perform some checking, and I will update accordingly.